commands/sketch/new.go

```diff
@@ -59,3 +69,20 @@ func NewSketch(ctx context.Context, req *rpc.NewSketchRequest) (*rpc.NewSketchRe
 
 	return &rpc.NewSketchResponse{MainFile: sketchMainFilePath.String()}, nil
 }
+
+func validateSketchName(name string) error {
+	if name == "" {
+		return &arduino.CantCreateSketchError{Cause: errors.New(tr("sketch name cannot be empty"))}
+	}
+	if len(name) > sketchNameMaxLength {
+		return &arduino.CantCreateSketchError{Cause: errors.New(tr("sketch name too long (%d characters). Maximum allowed length is %d",
+			len(name),
+			sketchNameMaxLength))}
+	}
+	if !sketchNameValidationRegex.MatchString(name) {
+		return &arduino.CantCreateSketchError{Cause: errors.New(tr("invalid sketch name \"%s\". Required pattern %s",
```

Comment:
This error outputs: `invalid sketch name "xxx". Required pattern ^[0-9a-zA-Z][0-9a-zA-Z_\.-]*$`
Not everyone is able to digest a regex, maybe a little explanation is better:
`invalid sketch name "xxx": the first character must be alphanumeric, the following ones can also contain "_", "-", and ".".`